magento/devdocs
# Make analytics fail gracefully

Problem:
Links such as "Edit this page on GitHub", "Give us feedback", "Become Contributor" cannot be followed if for some reason Google Analytics fails to load -- for example if a privacy tool such as Ghostery or Disconnect is installed in the browser (https://davidwalsh.name/detecting-google-analytics).

Changes:
- Checks GA is loaded and functioning before intercepting click events on links with `e.preventDefault` for analytics (analytics.js)
- Checks Munchkin is loaded successfully before trying to initiate it (footer-scripts.html)

Thoughts:
It's not ideal checking for GA in this way within each click event. It's unnecessary duplication and would be easy to mistakenly introduce more event tracking which would break in this scenario, without realising. Feel free to burn this PR for a more thought out solution! But failing that, this change should provide a better experience for the privacy minded developer. 

QA:
- Confirm `Munchkin` is initiated and Google Analytics click events are still fired under normal circumstances.
- Using a tool such as Ghostery or Disconnect block Google Analytics and Munchkin. Clicking links which usually have analytics tied to them such as  "Edit this page on GitHub" or "Give us feedback" follow through successfully without tracking (as they have been blocked by the client extensions).
